flink-dstl/flink-dstl-dfs/src/main/java/org/apache/flink/changelog/fs/ChangelogStreamHandleReaderWithCache.java

```diff
@@ -161,23 +161,28 @@ private FileInputStream openAndSeek(RefCountedFile refCountedFile, long offset)
 
     private DataInputStream wrapStream(Path dfsPath, FileInputStream fin) {
         return new DataInputStream(new BufferedInputStream(fin)) {
+            private boolean closed = false;
+
             @Override
             public void close() throws IOException {
-                try {
-                    super.close();
-                } finally {
-                    cache.computeIfPresent(
-                            dfsPath,
-                            (key, value) -> {
-                                value.release();
-                                if (value.getReferenceCounter() == NO_USING_REF_COUNT) {
-                                    cacheCleanScheduler.schedule(
-                                            () -> cleanCacheFile(dfsPath),
-                                            cacheIdleMillis,
-                                            TimeUnit.MILLISECONDS);
-                                }
-                                return value;
-                            });
+                if (!closed) {
+                    closed = true;
```

Comment:
The returned `InputStream` might be used by different Tasks, i.e. different threads, right?

So this flag should be thread-safe.
Probably something like
```
private AtomicBoolean closed = new AtomicBoolean(false);
...
if (closed.compareAndSet(false, true)) { 
  // do close
}
```
WDYT?

Reply:
Thanks for reply @rkhachatryan . In my understanding, although ChangelogStreamHandlereaderWithCache will be used in multiple tasks, InputStream returned by ChangelogStreamHandleReaderWithCache#openAndSeek() will only be used in a single task (single thread). Can you confirm that ?

And I will write a test for this case.

Reply:
Yes, you're right, thanks for the correction
(I've also checked whether it can be closed by a non-task thread (e.g. watchdog), but that's not the case).